benchmarks/cpp/tensorexpr/tensorexpr.cpp

```diff
@@ -61,7 +61,72 @@ BENCHMARK_DEFINE_F(Gemm, TensorExprNoopt)(benchmark::State& state) {
   }
 }
 
+BENCHMARK_DEFINE_F(Gemm, TensorExprTile32x32)(benchmark::State& state) {
+  te::KernelScope ks;
+
+  te::Placeholder AP(te::BufHandle("A", {M, K}, te::kFloat));
+  te::Placeholder BP(te::BufHandle("B", {K, N}, te::kFloat));
+  te::Tensor* CT = te::Reduce(
+      "gemm",
+      {{M, "M"}, {N, "N"}},
+      te::Sum(),
+      [&](const te::ExprHandle& m, const te::ExprHandle& n, const te::ExprHandle& k) {
+        return AP.load(m, k) * BP.load(k, n);
+      },
+      {{K, "K"}});
+  te::LoopNest loop({CT});
+
+  {
+    auto const& loops = loop.getLoopStmtsFor(CT);
+    te::For* m = loops[0];
+    te::For* mo;
+    te::For* mi;
+    loop.splitWithMask(m, 32, &mo, &mi);
```

Comment:
Could you add a comment here (and below) why this split is needed?

Reply:
Sure, what's the needed clarification?  Just like, why one would tile a matrix multiplication?

Reply:
I meant the mask version vs just split with tail, turns out I read too much into it per Nick's comment. I think it's ok as is.